src/Joystick/Joystick.cc

```diff
@@ -609,17 +653,60 @@ void Joystick::_handleAxis()
     float throttle = _adjustRange(_rgAxisValues[axis], _rgCalibration[axis], (_throttleMode == ThrottleModeDownZero) ? false :_deadband);
 
     float gimbalPitch = 0.0f;
+    float gimbalYaw = 0.0f;
+
     if (_axisCount > 4) {
         axis = _rgFunctionAxis[gimbalPitchFunction];
         gimbalPitch = _adjustRange(_rgAxisValues[axis], _rgCalibration[axis],_deadband);
     }
 
-    float gimbalYaw = 0.0f;
     if (_axisCount > 5) {
         axis = _rgFunctionAxis[gimbalYawFunction];
         gimbalYaw = _adjustRange(_rgAxisValues[axis],   _rgCalibration[axis],_deadband);
     }
 
+    if (_axisCount > 5 && _gimbalAxisEnabled) {
+        int pitchAxisIndex = 5;
+        float gimbalPitchNorm = _adjustRange(_rgAxisValues[pitchAxisIndex], _rgCalibration[pitchAxisIndex], _deadband);
+        // float gimbalPitchDeg = gimbalPitchNorm * 160.0f - 80.0f;
+        // float gimbalPitchDeg = gimbalPitchNorm * (2.0f * gimbalPitchMax) - gimbalPitchMax;
+        float gimbalPitchDeg = gimbalPitchNorm * (2.0f * static_cast<float>(_gimbalMaxSpeed)) - static_cast<float>(_gimbalMaxSpeed);
+        gimbalPitchDeg=gimbalPitchDeg-5;
+
+        float gimbalPitchDeadzone = static_cast<float>(_rgCalibration[5].deadband);
+        float gimbalPitchOut = applyDeadzone(gimbalPitchDeg, gimbalPitchDeadzone);
+
+        int yawAxisIndex = 4;
+        float gimbalYawNorm = _adjustRange(_rgAxisValues[yawAxisIndex], _rgCalibration[yawAxisIndex], _deadband);
+        // float gimbalYawDeg = gimbalYawNorm * 160.0f - 80.0f;
+        float gimbalYawDeg = gimbalYawNorm * (2.0f * static_cast<float>(_gimbalMaxSpeed)) - static_cast<float>(_gimbalMaxSpeed);
+        gimbalYawDeg=gimbalYawDeg-4;
+
+        float gimbalYawDeadzone = static_cast<float>(_rgCalibration[4].deadband);
+        float gimbalYawOut = applyDeadzone(gimbalYawDeg, gimbalYawDeadzone);
+
+        static int zeroPitchCount = 0;
+        static int zeroYawCount = 0;
+
+        if (std::abs(gimbalPitchOut) == 0) {
+            zeroPitchCount++;
+        } else {
+            zeroPitchCount = 0; 
+        }
+
+        if (std::abs(gimbalYawOut) == 0) {
+            zeroYawCount++;
+        } else {
+            zeroYawCount = 0; 
+        }
+
+        if (zeroPitchCount >= 3 && zeroYawCount >= 3) {
+
+        }else{
+            _activeVehicle->sendGimbalAbsolutePosition(gimbalPitchOut, gimbalYawOut);
+            // qDebug() << gimbalYawOut << gimbalPitchOut;
+        }
```

Comment:
This is a bit weird. Can you add a comment how it works. And also remove avoid the empty if clause.

And don't use `static` there but instead it should be a class member variable.